python/ray/train/tensorflow/tensorflow_checkpoint.py

```diff
@@ -48,9 +70,181 @@ def from_model(
+
+        The path must maintain valid even after this function returns.
+
+        Args:
+            dir_path: The directory containing the saved model. This is the same
+                directory as used by ``model.save(dir_path)``.
+            preprocessor: A fitted preprocessor to be applied before inference.
+
+        Returns:
+            A :py:class:`TensorflowCheckpoint` converted from SavedModel format.
+
+        Examples:
+
+            >>> import tensorflow as tf
+
+            >>> import ray
+            >>> from ray.train.batch_predictor import BatchPredictor
+            >>> from ray.train.tensorflow import (
+            ... TensorflowCheckpoint, TensorflowTrainer, TensorflowPredictor)
+            >>> from ray.air import session
+            >>> from ray.air.config import ScalingConfig
+
+            >>> def train_fn():
+            ...     model = tf.keras.Sequential(
+            ...         [
+            ...             tf.keras.layers.InputLayer(input_shape=()),
+            ...             tf.keras.layers.Flatten(),
+            ...             tf.keras.layers.Dense(10),
+            ...             tf.keras.layers.Dense(1),
+            ...         ])
+            ...     model.save("my_model")
+            ...     checkpoint = TensorflowCheckpoint.from_saved_model("my_model")
+            ...     session.report({"my_metric": 1}, checkpoint=checkpoint)
+
+            >>> trainer = TensorflowTrainer(
+            ...     train_loop_per_worker=train_fn,
+            ...     scaling_config=ScalingConfig(num_workers=2))
+
+            >>> result_checkpoint = trainer.fit().checkpoint
+
+            >>> batch_predictor = BatchPredictor.from_checkpoint(
+            ...     result_checkpoint, TensorflowPredictor)
+            >>> batch_predictor.predict(ray.data.range(3))
+        """
+        if preprocessor:
+            save_preprocessor_to_dir(preprocessor, dir_path)
+        checkpoint = cls.from_directory(dir_path)
+        checkpoint._flavor = cls.Flavor.SAVED_MODEL
         return checkpoint
 
-    def get_model_weights(self) -> tf.keras.Model:
-        """Retrieve the model weights stored in this checkpoint."""
-        model_weights, _ = _load_checkpoint(self, "TensorflowTrainer")
-        return model_weights
+    def get_model(
+        self,
+        model_definition: Optional[
+            Union[Callable[[], tf.keras.Model], Type[tf.keras.Model]]
+        ] = None,
+    ) -> tf.keras.Model:
+        """Retrieve the model stored in this checkpoint.
+
+        Args:
+            model_definition: This arg is expected if and only if
+                the checkpoint's flavor is `MODEL_WEIGHTS`.
+
+        Returns:
+            A Tensorflow Keras model.
+        """
+        if self._flavor is self.Flavor.MODEL_WEIGHTS:
+            if not model_definition:
+                raise ValueError(
+                    "Expecting input of `model` argument when checkpoint's flavor "
+                    "is `MODEL_WEIGHTS`."
```

Comment:
Can we have a better error message here? As a user I'm not sure what flavor means. Also there is no `model` argument- it's `model_definition`

Reply:
can we also add a test for this?